src/controllers/__tests__/events.test.js

```diff
@@ -96,3 +96,69 @@ describe("Testing events for routes doesn't require auth controls", () => {
       });
   });
 });
+
+describe('Testing events for routes require auth controls', () => {
+  test('POST /api/events should create a new event and return it in the response', (done) => {
+    const req = {
+      user: {
+        id: user1.id,
+      },
+    };
+    request(app)
+      .post('/api/events')
+      .set('Content-Type', 'application/json')
+      .send(events[0])
+      .expect('Content-Type', /json/)
+      .expect(201, (err, res) => {
+        if (err) {
+          done();
+          return err;
+        }
+        expect(req.user.id).toBe(ngos.id);
+        expect(res.body.data.avatar).toEqual(events[0].avatar);
+        expect(res.body.data.description).toEqual(events[0].description);
+        expect(res.body.data.location).toEqual(events[0].location);
+        expect(res.body.data.launchDate).toEqual(events[0].launchDate);
+        expect(res.body.data.ngoId).toEqual(events[0].ngo);
+        expect(res.body.data.topic).toEqual(events[0].topic);
+        done();
+        return events[0];
+      });
+  });
+
+  test('POST /api/events/ should refuse to add event without authentication', (done) => {
+    const req = {};
+    request(app)
+      .post('/api/events/')
+      .expect('Content-Type', 'application/json; charset=utf-8')
+      .expect({
+        message: 'You need to sign in to add an event',
+      })
+      .expect(403, (err, res) => {
+        if (err) {
+          done();
+          return err;
+        }
+        expect(req.user.id).not.toBe(ngos.id);
+        done();
+        return events[0];
+      });
+    done();
+  });
+
+  test('POST /api/events/ should refuse to add event without authorization', (done) => {
+    const req = {
+      user: {
+        id: user2.id,
+      },
+    };
+    request(app)
+      .post('/api/events/')
+      .expect(403)
+      .expect('Content-Type', 'application/json; charset=utf-8')
+      .expect({
+        message: 'You are not authorized to delete this event',
```

Comment:
The message says: 'You are not authorized to delete this event'. delete or add an event? 🤣